test/uk/gov/hmrc/helptosavefrontend/connectors/HelpToSaveConnectorSpec.scala

```diff
@@ -19,63 +19,85 @@ package uk.gov.hmrc.helptosavefrontend.connectors
 import cats.data.EitherT
 import cats.instances.int._
 import cats.syntax.eq._
+import com.typesafe.config.ConfigFactory
+import org.mockito.IdiomaticMockito
 import org.scalacheck.Gen
+import org.scalatest.EitherValues
+import org.scalatestplus.play.guice.GuiceOneAppPerSuite
 import org.scalatestplus.scalacheck.ScalaCheckDrivenPropertyChecks
+import play.api.inject.guice.GuiceApplicationBuilder
+import play.api.{Application, Configuration}
 import play.api.libs.json._
 import uk.gov.hmrc.helptosavefrontend.connectors.HelpToSaveConnectorImpl.GetEmailResponse
-import uk.gov.hmrc.helptosavefrontend.controllers.ControllerSpecWithGuiceApp
+import uk.gov.hmrc.helptosavefrontend.controllers.{ControllerSpecBase, ControllerSpecWithGuiceApp}
 import uk.gov.hmrc.helptosavefrontend.models.TestData.UserData.validNSIPayload
 import uk.gov.hmrc.helptosavefrontend.models._
 import uk.gov.hmrc.helptosavefrontend.models.account.{Account, AccountNumber}
 import uk.gov.hmrc.helptosavefrontend.models.eligibility.EligibilityCheckResultType.{AlreadyHasAccount, Eligible, Ineligible}
 import uk.gov.hmrc.helptosavefrontend.models.eligibility.{EligibilityCheckResponse, EligibilityCheckResult}
 import uk.gov.hmrc.helptosavefrontend.models.register.CreateAccountRequest
 import uk.gov.hmrc.helptosavefrontend.services.HelpToSaveServiceImpl.SubmissionSuccess
-import uk.gov.hmrc.http.HttpResponse
+import uk.gov.hmrc.helptosavefrontend.util.WireMockMethods
+import uk.gov.hmrc.http.{HeaderCarrier, HttpResponse}
+import uk.gov.hmrc.http.test.WireMockSupport
 
 import java.time.LocalDate
 import java.util.{Base64, UUID}
-import scala.concurrent.Future
+import scala.concurrent.{ExecutionContext, Future}
 
 // scalastyle:off magic.number
-class HelpToSaveConnectorSpec extends ControllerSpecWithGuiceApp with HttpSupport with ScalaCheckDrivenPropertyChecks {
+class HelpToSaveConnectorSpec
+    extends ControllerSpecBase with IdiomaticMockito with WireMockSupport with WireMockMethods with GuiceOneAppPerSuite
+    with EitherValues with ScalaCheckDrivenPropertyChecks {
 
-  lazy val connector: HelpToSaveConnector = new HelpToSaveConnectorImpl(mockHttp)
+  val (desBearerToken, desEnvironment) = "token" -> "environment"
+  private val config = Configuration(
+    ConfigFactory.parseString(
+      s"""
+         |microservice {
+         |  services {
+         |      help-to-save {
+         |      protocol = http
+         |      host     = $wireMockHost
+         |      port     = $wireMockPort
+         |    }
+         |  }
+         |}
+         |
+         |des {
```

Comment:
this config probably doesn't need `des` override. Can remove

Reply:
Done